src/test/java/io/strimzi/kafka/bridge/http/HttpCorsTests.java

```diff
@@ -0,0 +1,193 @@
+    static {
+        config.put(HttpConfig.HTTP_CONSUMER_TIMEOUT, timeout);
+        config.put(BridgeConfig.BRIDGE_ID, "my-bridge");
+    }
+
+    static Vertx vertx;
+    static HttpBridge httpBridge;
+    static WebClient client;
+
+    static BridgeConfig bridgeConfig;
+    static KafkaFacade kafkaCluster = new KafkaFacade();
+
+    @BeforeAll
+    static void beforeAll() {
+        kafkaCluster.start();
+    }
+
+    @BeforeEach
+    void prepare() {
+        VertxOptions options = new VertxOptions();
+        options.setMaxEventLoopExecuteTime(Long.MAX_VALUE);
+        vertx = Vertx.vertx(options);
+    }
+
+    @AfterEach
+    void cleanup() {
+        vertx.close();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        kafkaCluster.stop();
+    }
+
+    @Test
+    public void testCorsNotEnabled(VertxTestContext context) {
+        createWebClient(false, null, context);
+
+        if (!"TRUE".equalsIgnoreCase(System.getenv("STRIMZI_USE_SYSTEM_BRIDGE"))) {
+            vertx.deployVerticle(httpBridge, context.succeeding(id -> client
+                .request(HttpMethod.OPTIONS, 8080, "localhost", "/ready")
+                .putHeader("Origin", "https://evil.io")
+                .putHeader("Access-Control-Request-Method", "GET")
+                .send(ar -> {
+                    context.verify(() -> {
+                        assertThat(ar.result().statusCode(), is(405));
+                        context.completeNow();
+                    });
+
+                })));
+
+        } else {
+            context.completeNow();
+        }
+    }
+
+    @Test
+    public void testOriginForbiddenRequest(VertxTestContext context) {
+        createWebClient(true, null, context);
+
+        if (!"TRUE".equalsIgnoreCase(System.getenv("STRIMZI_USE_SYSTEM_BRIDGE"))) {
+            vertx.deployVerticle(httpBridge, context.succeeding(id -> client
+                    .request(HttpMethod.OPTIONS, 8080, "localhost", "/ready")
+                    .putHeader("Origin", "https://evil.io")
+                    .putHeader("Access-Control-Request-Method", "GET")
+                    .send(ar -> {
+                        context.verify(() -> {
+                            assertThat(ar.result().statusCode(), is(403));
+                            context.completeNow();
+                        });
+
+                    })));
+
+        } else {
+            context.completeNow();
+        }
+    }
+
+    @Test
+    public void testOriginAllowedRequest(VertxTestContext context) {
+        createWebClient(true, null, context);
+        final String origin = "https://strimzi.io";
+
+        if (!"TRUE".equalsIgnoreCase(System.getenv("STRIMZI_USE_SYSTEM_BRIDGE"))) {
+            vertx.deployVerticle(httpBridge, context.succeeding(id -> client
+                    .request(HttpMethod.OPTIONS, 8080, "localhost", "/ready")
+                    .putHeader("Origin", "https://strimzi.io")
+                    .putHeader("Access-Control-Request-Method", "GET")
```

Comment:
out of curiosity and genuine question ...
You are actually making an `OPTIONS` HTTP request specifying the `GET` just as part of the header.
Because the CORS is enabled in this test, is the corresponding `GET` automatically performed or just the `OPTIONS`? My guess is that no `GET` is executed. Should we test that after getting an "ok" we should run the actual `GET`?

Reply:
yes it is executed only the OPTIONS; I will add a GET after the OPTIONS ok

Reply:
if you perform a GET request without a preflight, you will get a 200 OK; there is no control at that level

Reply:
so what kind of CORS is that? Any client can just perform a GET without preflight OPTIONS and, even with CORS disabled, it works?

Reply:
yes, browsers are instructed to do this, to perform a preflgiht

Reply:
Well for POST requests, without a preflight, you will have a 403.
For GET requests no, you will still get a 200